.github/workflows/update-beats.yml

```
@@ -1,20 +1,29 @@
 name: update-beats
 
 on:
-  push:
-    branches:
-      - main
-      - "7.17"
-      - "8.*"
-      - '!8.[0-5]'
+  workflow_dispatch:
+  schedule:
+    - cron: '0 15 * * 1-5'
 
 permissions:
   pull-requests: write
   contents: write
 
 jobs:
+  filter:
+    runs-on: ubuntu-latest
+    timeout-minutes: 1
+    outputs:
+      matrix: ${{ steps.generator.outputs.matrix }}
+    steps:
+      - id: generator
+        uses: elastic/apm-pipeline-library/.github/actions/elastic-stack-snapshot-branches@current
   bump:
+    needs:
+      - filter
     runs-on: ubuntu-latest
+    strategy:
+      matrix: ${{ fromJson(needs.filter.outputs.matrix) }}
```

Override matrix without failing fast (otherwise it's defaulted to true), https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#jobsjob_idstrategyfail-fast

```suggestion
      fail-fast: false
      matrix: ${{ fromJson(needs.filter.outputs.matrix) }}
```

---

Ah. thanks!.. forgot about that.

---

https://github.com/elastic/apm-server/pull/10244/commits/20bb5a4a721f9bbfb3d3843092052628fad41508